kernel/kernel-api/src/main/java/io/delta/kernel/internal/SnapshotImpl.java

```diff
@@ -75,14 +96,30 @@ public ScanBuilder getScanBuilder(Engine engine) {
     return new ScanBuilderImpl(dataPath, protocol, metadata, getSchema(engine), logReplay, engine);
   }
 
-  public Metadata getMetadata() {
-    return metadata;
+  ///////////////////
+  // Internal APIs //
+  ///////////////////
+
+  public Path getLogPath() {
```

Comment:
1. put all getters at the top of internal APIs, with more complicated internal APIs below
2. re-ordered these getters to match the order of member field declarations